### PR TITLE
Fix/datepicker

### LINKED ISSUE #229: Layout.hs has missing JS/source files
The generated `Web/View/Layout.hs` file does not included all the required JS/CSS vendor files:

* flatpickr.js
* flatpickr.min.css
* select2.min.css and select2.js ?
* prod.js is not used (all vendor JS files are included individually)

### PR DESCRIPTION
**Pull request by nghamilton**
Fixes #229

### CODE REVIEW

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 1]
```diff
@@ -118,9 +118,7 @@ generateGenericApplication applicationName =
                 <> "<head>\n"
                 <> "    {metaTags}\n"
                 <> "\n"
-                <> "    <link rel=\"stylesheet\" href=\"/vendor/bootstrap.min.css\"/>\n"
-                <> "    <link rel=\"stylesheet\" href=\"/app.css\"/>\n"
-                <> "\n"
+                <> "    {cssfiles}\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:121]: Could we call this `stylesheets`? :)

```suggestion
                <> "    {stylesheets}\n"
```

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 2]
```diff
@@ -133,22 +131,28 @@ generateGenericApplication applicationName =
                 <> "</body>\n"
                 <> "|]\n"
                 <> "\n"
+                <> "cssfiles = do\n"
+                <> "    when (isDevelopment FrameworkConfig.environment) [hsx|\n"
+                <> "      <link rel=\"stylesheet\" href=\"/vendor/bootstrap.min.css\"/>\n"
+                <> "      <link rel=\"stylesheet\" href=\"/vendor/flatpickr.min.css\"/>\n"
+                <> "    |]\n"
+                <> "    [hsx| <link rel=\"stylesheet\" href=\"/app.css\"/> |]\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:139]: Usually the `app.css` is not the bundled css. [There is the `prod.css` file which can be built using `make`](https://github.com/digitallyinduced/ihp/blob/master/lib/IHP/Makefile.dist#L134). I guess we need to include the `prod.css` here in production mode, and the `app.css` only in dev.

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 3]
```diff
@@ -133,22 +131,28 @@ generateGenericApplication applicationName =
...
                 <> "    |]\n"
                 <> "    when (isProduction FrameworkConfig.environment) [hsx|\n"
-                <> "            <script src=\"/vendor/turbolinks.js\"></script>\n"
-                <> "            <script src=\"/vendor/morphdom-umd.min.js\"></script>\n"
-                <> "            <script src=\"/vendor/turbolinksMorphdom.js\"></script>\n"
-                <> "            <script src=\"/vendor/turbolinksInstantClick.js\"></script>\n"
-                <> "        |]\n"
+                <> "       <script src=\"/prod.js\"></script>\n"
+                <> "    |]\n"
+                <> "    [hsx| <script src=\"/helpers-extra.js\"></script> |]\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:155]: What is the idea behind the `helpers-extra` file? :)

File: IHP/View/Form.hs [hunk 4]
```diff
@@ -390,7 +390,7 @@ dateField :: forall fieldName model value.
     , InputValue value
     , KnownSymbol (GetModelName model)
     ) => Proxy fieldName -> FormField
-dateField field = (textField field) { fieldType = DateInput }
+dateField field = (textField field) { fieldType = DateInput, fieldClass = "date-field" }
```
Comment (mpscholten) [inline IHP/View/Form.hs:393]: Why are we changing this?

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 5]
```diff
@@ -133,22 +131,30 @@ generateGenericApplication applicationName =
                 <> "</body>\n"
                 <> "|]\n"
                 <> "\n"
+                <> "stylesheets = do\n"
+                <> "    when (isDevelopment FrameworkConfig.environment) [hsx|\n"
+                <> "      <link rel=\"stylesheet\" href=\"/vendor/bootstrap.min.css\"/>\n"
+                <> "      <link rel=\"stylesheet\" href=\"/vendor/flatpickr.min.css\"/>\n"
+                <> "      <link rel=\"stylesheet\" href=\"/app.css\"/>\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:138]: I think we need to add two more spaces here, as all other code is also indented with 4 spaces :)

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 6]
```diff
@@ -133,22 +131,30 @@ generateGenericApplication applicationName =
...
+                <> "    |]\n"
+                <> "    when (isProduction FrameworkConfig.environment) [hsx|\n"
+                <> "      <link rel=\"stylesheet\" href=\"/prod.css\"/>\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:141]: same here

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 7]
```diff
@@ -133,22 +131,30 @@ generateGenericApplication applicationName =
...
-                <> "        <script src=\"/helpers.js\"></script>\n"
+                <> "    when (isDevelopment FrameworkConfig.environment) [hsx|\n"
+                <> "      <script id=\"livereload-script\" src=\"/livereload.js\"></script>\n"
+                <> "      <script src=\"/vendor/jquery-3.2.1.slim.min.js\"></script>\n"
+                <> "      <script src=\"/vendor/timeago.js\"></script>\n"
+                <> "      <script src=\"/vendor/popper.min.js\"></script>\n"
+                <> "      <script src=\"/vendor/bootstrap.min.js\"></script>\n"
+                <> "      <script src=\"/vendor/flatpickr.js\"></script>\n"
+                <> "      <script src=\"/helpers.js\"></script>\n"
+                <> "      <script src=\"/vendor/morphdom-umd.min.js\"></script>\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:153]: also here

File: IHP/IDE/CodeGen/ApplicationGenerator.hs [hunk 8]
```diff
@@ -133,22 +131,30 @@ generateGenericApplication applicationName =
...
                 <> "    |]\n"
                 <> "    when (isProduction FrameworkConfig.environment) [hsx|\n"
-                <> "            <script src=\"/vendor/turbolinks.js\"></script>\n"
-                <> "            <script src=\"/vendor/morphdom-umd.min.js\"></script>\n"
-                <> "            <script src=\"/vendor/turbolinksMorphdom.js\"></script>\n"
-                <> "            <script src=\"/vendor/turbolinksInstantClick.js\"></script>\n"
-                <> "        |]\n"
+                <> "       <script src=\"/prod.js\"></script>\n"
```
Comment (mpscholten) [inline IHP/IDE/CodeGen/ApplicationGenerator.hs:156]: and here :)

Comment (nghamilton) [reply IHP/IDE/CodeGen/ApplicationGenerator.hs]: Shocker. Sorry, this PR has been a mess.

**Pull request merged**